contracts/LifToken.sol

```diff
@@ -457,14 +469,20 @@ contract LifToken is Ownable, ERC20, SafeMath, PullPayment {
         giveVotes(msg.sender, to);
       }
 
-      TransferData(msg.sender, to, value, data);
+      if (doCall && to.call(data))
+        TransferData(msg.sender, to, value, data);
+      else if (!doCall)
+        TransferData(msg.sender, to, value, data);
 
       return true;
 
     }
 
     // ERC20 transferFrom method but with data parameter.
-    function transferDataFrom(address from, address to, uint value, string data) onStatus(3,4) returns (bool success) {
+    function transferDataFrom(address from, address to, uint value, bytes data, bool doCall) onStatus(3,4) returns (bool success) {
+
+      if (to == address(this))
```

Comment:
should we extract this check to a fn like `checkNotSelfTransfer`? I'm ok if it's in a further PR